mmv1/products/privateca/api.yaml

```diff
@@ -108,107 +117,248 @@ objects:
+                    - !ruby/object:Api::Type::NestedObject
+                      name: 'objectId'
+                      description: |
+                        Describes values that are relevant in a CA certificate.
+                      properties:
+                        - !ruby/object:Api::Type::Array
+                          name: 'objectIdPath'
+                          required: true
+                          item_type: Api::Type::Integer
+                          description: |
+                            An ObjectId specifies an object identifier (OID). These provide context and describe types in ASN.1 messages.
+              - !ruby/object:Api::Type::Array
+                name: 'policyIds'
+                description: |
+                  Describes the X.509 certificate policy object identifiers, per https://tools.ietf.org/html/rfc5280#section-4.2.1.4.
+                item_type: !ruby/object:Api::Type::NestedObject
+                  properties:
+                    - !ruby/object:Api::Type::Array
+                      name: 'objectIdPath'
+                      required: true
+                      item_type: Api::Type::Integer
+                      description: |
+                        An ObjectId specifies an object identifier (OID). These provide context and describe types in ASN.1 messages.
+              - !ruby/object:Api::Type::Array
+                name: 'aiaOcspServers'
+                item_type: Api::Type::String
+                description: |
+                  Describes Online Certificate Status Protocol (OCSP) endpoint addresses that appear in the
+                  "Authority Information Access" extension in the certificate.
+              - !ruby/object:Api::Type::NestedObject
+                name: 'caOptions'
+                description: |
+                  Describes values that are relevant in a CA certificate.
+                properties:
+                  - !ruby/object:Api::Type::Boolean
+                    name: 'isCa'
+                    description: |
+                      Refers to the "CA" X.509 extension, which is a boolean value. When this value is missing,
+                      the extension will be omitted from the CA certificate.
+                  - !ruby/object:Api::Type::Integer
+                    name: 'maxIssuerPathLength'
+                    description: |
+                      Refers to the path length restriction X.509 extension. For a CA certificate, this value describes the depth of
+                      subordinate CA certificates that are allowed. If this value is less than 0, the request will fail. If this
+                      value is missing, the max path length will be omitted from the CA certificate.
+              - !ruby/object:Api::Type::NestedObject
+                name: 'keyUsage'
+                description: |
+                  Optional. Indicates the intended use for keys that correspond to a certificate.
+                properties:
```

Comment:
Are these subobjects not required when the parent is set? Same w/ their subfields

Reply:
Not technically, but due to API behavior not returning these when all sub-values are set to false I have added a custom flattener & marked all of these subobjects as required.

I think this is a reasonable way to handle the tricky situation where these fields have slightly different behaviors between the 3 PrivateCA resources although the structure of the objects is almost exactly identical. We could also look into customizediffs, or not allowing objects with all-false values to prevent this in the future without marking these as required

Reply:
Well, some of them are actually required depending on the resource.

For example, `caOptions.isCa` must be set to true for a CertificateAuthority resource.